### DISCUSSION TITLE
Keyboard brightness

### DISCUSSION THREAD

**Discussion by Biziame** [seerge/g-helper#181] [Ideas]
Is it possible to add a function that allows to choose the number of minutes before the laptop’s brightness turns off when it is powered by the battery?

**Comment by seerge** [comment]
I think on hardware level, it's set to something like 10 seconds and can't be changed.

**Reply by Biziame** [reply]
https://www.reddit.com/r/ZephyrusG14/comments/j4ip6f/how_to_stop_keyboard_lighting_from_automatically/

This solution is for the previous model, but I believe it can also be done in the latest model. By default, the timer is set to turn off after one minute.
Is it possible to implement the function directly in the software, choosing from time to time whether to keep the function active and for how many minutes, since the file can be modified?

**Reply by seerge** [reply]
@Biziame ok, so actually it's software controlled by Asus Optimization Service.
Main problem here - is that to edit that registry value - app will need administrator privileges (that it doesn't use at the moment). Forcing whole app for all users running as admin only for a small flag in registry is too much imho.

**Reply by Biziame** [reply]
@seerge I noticed in the latest software release that administrator privileges are now available. Would it be possible to add a feature that allows users to choose how long they want to keep the keyboard backlight on?

**Reply by seerge** [reply]
[GHelper.zip](https://github.com/seerge/g-helper/files/11443989/GHelper.zip)
@Biziame Try this bild, it should ask for permissions when needed , and if you are already admin - it will change registry

**Reply by Biziame** [reply]
> [GHelper.zip](https://github.com/seerge/g-helper/files/11443989/GHelper.zip) @Biziame Try this bild, it should ask for permissions when needed , and if you are already admin - it will change registry

I tried the build you posted in the other thread. I can tell you that it works, but it doesn't allow me to set more than 100 seconds

**Reply by seerge** [reply]
@Biziame that was a default limit for that ui component :) I have increased it to 3600 (1 hour) in https://github.com/seerge/g-helper/releases/tag/v0.60

**Comment by chege54** [comment]
A question regarding the keyboard backlit level. If I drag the slider, is that should change the brightless level if I turned it off manually?
I have GA503RW but if I drag it does nothing,
(I mean, if the backlit is tuned off via the FN+F2 I can not turn on it with the slider. Is that a normal behavior?

**Reply by seerge** [reply]
Fn+F2 also just change brightness, not "turn off" backlight. So slider should set brightness back.

**Reply by chege54** [reply]
But in my case, the slider does nothing.
Do I try to support you find the root casue?

**Reply by seerge** [reply]
and does fn+f2/f3 work for you? you can post a log from app here to see if app sees your keyboard

**Reply by chege54** [reply]
fn+f2 anf f3 changes the brightness,
![log](https://github.com/seerge/g-helper/assets/5606998/f55788ab-cad1-4834-83b3-5479e2bc9504)

that is the log if I am dragging the slider back and forths.
It is white backlit version only model

This is log when I am changeing the backlit via fn+F2 F3
![log2](https://github.com/seerge/g-helper/assets/5606998/47a80810-86e1-4334-afc3-8524620ff956)

**Reply by seerge** [reply]
@chege54 well, app tries to send a brightness setting to  keyboard, i don't know why it's being ignored. Does OpenRGB able to control brightness in your case?

And generally, why do you need it slider? I was initally added for a guy who couldn't control keyboard with hotkeys as a workaround :)) for rest of the users hotkeys are way more convenient :)

**Reply by chege54** [reply]
I did not try with OpenRGB yet, but probably I will. 
Not an essential feature, but If the room is totally dark and I am not seeing the proper F key, :) it seemd to me easier to turn it on

**Reply by chege54** [reply]
But actually the OpenRGB does nothing also

**Reply by seerge** [reply]
Ok, then i definitely can’t help here, as i don’t have device + it’s the only open source information that is available
